desc/integrals/basis.py

```diff
@@ -58,6 +85,621 @@ def _in_epigraph_and(is_intersect, df_dy_sign, /):
     return put(is_intersect, idx[0], edge_case)
 
 
+def _chebcast(cheb, arr):
+    """Add leftmost axis to ``cheb`` depending on ``arr.ndim``.
+
+    Input ``arr`` should not have rightmost dimension of cheb that iterates
+    coefficients, but may have additional leftmost dimension for batch operation.
+    """
+    errorif(
+        jnp.ndim(arr) > cheb.ndim,
+        NotImplementedError,
+        msg=f"Only one additional axis for batch dimension is allowed. "
+        f"Got {jnp.ndim(arr) - cheb.ndim + 1} additional axes.",
+    )
+    return cheb if jnp.ndim(arr) < cheb.ndim else cheb[jnp.newaxis]
+
+
+class FourierChebyshevSeries(IOAble):
+    """Real-valued Fourier-Chebyshev series.
+
+    f(x, y) = ∑ₘₙ aₘₙ ψₘ(x) Tₙ(y)
+    where ψₘ are trigonometric polynomials on [0, 2π]
+    and Tₙ are Chebyshev polynomials on [−yₘᵢₙ, yₘₐₓ].
+
+    Examples
+    --------
+    Let the magnetic field be B = ∇ρ × ∇x. This basis will then parameterize
+    maps in Clebsch coordinates. Passing in a sequence of x values tracking
+    the field line (see ``get_fieldline``) to the ``compute_cheb`` method will
+    generate a 1D parameterization of f along the field line.
+
+    This is useful to interpolate f ≝ θ and use the map x, ζ ↦ θ(x, ζ) to
+    compute quantities along field lines via evaluating Fourier series
+    parameterized in DESC computational coordinates θ, ζ, where the Fourier
+    transform is more condensed when NFP > 1.
+
+    Notes
+    -----
+    Performance may improve if ``X`` and ``Y`` are powers of two.
+
+
+    Parameters
+    ----------
+    f : jnp.ndarray
+        Shape (..., X, Y).
+        Samples of real function on the ``FourierChebyshevSeries.nodes`` grid.
+    domain : tuple[float]
+        Domain for y coordinates. Default is [-1, 1].
+    lobatto : bool
+        Whether ``f`` was sampled on the Gauss-Lobatto (extrema-plus-endpoint)
+        instead of the interior roots grid for Chebyshev points.
+
+    Attributes
+    ----------
+    X : int
+        Fourier spectral resolution.
+    Y : int
+        Chebyshev spectral resolution.
+
+    """
+
+    def __init__(self, f, domain=(-1, 1), lobatto=False):
+        """Interpolate Fourier-Chebyshev series to ``f``."""
+        errorif(domain[0] > domain[-1], msg="Got inverted domain.")
+        errorif(lobatto, NotImplementedError, "JAX hasn't implemented type 1 DCT.")
+        self.X = f.shape[-2]
+        self.Y = f.shape[-1]
+        self.domain = domain
+        self.lobatto = lobatto
+        self._c = rfft(
```

Comment:
because I am shameless in my ignorance, these are the a_mn?

which are fitting the inputted function values?

Reply:
those coefficients would be what are returned by harmonics method. these are the compactified form of the fourier chebyshev coefficients under the complex exponential and cosines.